docs/ref/models/constraints.txt

```diff
@@ -283,26 +283,38 @@ PostgreSQL 15+.
 .. attribute:: UniqueConstraint.violation_error_code
 
 The error code used when ``ValidationError`` is raised during
-:ref:`model validation <validating-objects>`. Defaults to ``None``.
+:ref:`model validation <validating-objects>`. The error code of such :class:`~UniqueConstraint`\s 
+will then be different from :attr:`.Field.unique` or in
+:attr:`Meta.unique_together <django.db.models.Options.unique_together>` if a custom 
+violation_error_code is set. Defaults to
+:attr:`.BaseConstraint.violation_error_message`.
+
+.. versionchanged:: 5.2
 
-This code is *not used* for :class:`UniqueConstraint`\s with
+This code *is* also used for :class:`UniqueConstraint`\s with
 :attr:`~UniqueConstraint.fields` and without a
-:attr:`~UniqueConstraint.condition`. Such :class:`~UniqueConstraint`\s have the
-same error code as constraints defined with :attr:`.Field.unique` or in
-:attr:`Meta.unique_together <django.db.models.Options.constraints>`.
+:attr:`~UniqueConstraint.condition`. The error code of such :class:`~UniqueConstraint`\s 
+will then be different from :attr:`.Field.unique` or in
+:attr:`Meta.unique_together <django.db.models.Options.unique_together>` if a custom 
+violation_error_code is set.
 
 ``violation_error_message``
 ---------------------------
 
 .. attribute:: UniqueConstraint.violation_error_message
 
 The error message used when ``ValidationError`` is raised during
-:ref:`model validation <validating-objects>`. Defaults to
+:ref:`model validation <validating-objects>`. The error message of such 
+:class:`~UniqueConstraint`\s will then be different from constraints defined 
+with :attr:`.Field.unique` or in :attr:`Meta.unique_together 
+<django.db.models.Options.unique_together>`. Defaults to
 :attr:`.BaseConstraint.violation_error_message`.
```

Comment:
```suggestion
The error message used when a ``ValidationError`` is raised during
:ref:`model validation <validating-objects>`.

If :attr:`~UniqueConstraint.fields` is set and without a
:attr:`~UniqueConstraint.condition`, the
:attr:`~django.db.models.UniqueConstraint.violation_error_message` defaults to
the same error message as when setting :attr:`.Field.unique` when there is a
single field, and to the same error message as when setting
:attr:`Meta.unique_together <django.db.models.Options.unique_together>` when
there are multiple.

Otherwise, defaults to :attr:`.BaseConstraint.violation_error_message`.
```